Update regex refs from upstream

- I automated regex building inside CI (during SNAP package creation). 
- I already created an issue in regex upstream asking for a release / git tag for some kind of stable version (so I can use an stable version, not head, during build).
- meanwhile, I guess JtR needs to update its regex references.

```
regex.c: In function ‘do_regex_hybrid_crack’:
regex.c:218:33: warning: passing argument 2 of ‘c_regex_cb’ from incompatible pointer type [-Wincompatible-pointer-types]
   regex_ptr = c_regex_cb(regex, callback);
                                 ^
In file included from regex.h:35:0,
                 from regex.c:12:
/home/claudio/fontes/packages/john-the-ripper/stage/include/librexgen/c/librexgen.h:40:13: note: expected ‘callback_fp {aka long unsigned int (*)(int *, const long unsigned int)}’ but argument is of type ‘size_t (*)(char *, const size_t) {aka long unsigned int (*)(char *, const long unsigned int)}’
 c_regex_ptr c_regex_cb(
             ^
regex.c: In function ‘do_regex_crack’:
regex.c:328:32: warning: passing argument 2 of ‘c_regex_cb’ from incompatible pointer type [-Wincompatible-pointer-types]
  regex_ptr = c_regex_cb(regex, callback);
                                ^
In file included from regex.h:35:0,
                 from regex.c:12:
/home/claudio/fontes/packages/john-the-ripper/stage/include/librexgen/c/librexgen.h:40:13: note: expected ‘callback_fp {aka long unsigned int (*)(int *, const long unsigned int)}’ but argument is of type ‘size_t (*)(char *, const size_t) {aka long unsigned int (*)(char *, const long unsigned int)}’
 c_regex_ptr c_regex_cb(
             ^
ar: creating aes.a
```

--- a/src/regex.c
+++ b/src/regex.c
@@ -138,7 +138,7 @@ static void rexgen_setlocale()
 // Would be nice to have SOME way to be thread safe!!!
 static char BaseWord[1024];
 
-size_t callback(char* dst, const size_t buffer_size)
+size_t callback(wchar_t* dst, const size_t buffer_size)
 {
 	int len;
 

